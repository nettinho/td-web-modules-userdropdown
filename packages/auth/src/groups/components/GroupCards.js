import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Icon, Card } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import routes from "../routes";
import GroupCard from "./GroupCard";

export const GroupCards = ({ groups }) => (
  <Card.Group>
    <Card href={routes.GROUP_CREATE}>
      <Card.Content textAlign="center">
        <Card.Header>
          <FormattedMessage id="groups.actions.create" />
        </Card.Header>
        <Card.Description>
          <Icon floated="center" name="add" size="massive" />
        </Card.Description>
      </Card.Content>
    </Card>
    {groups.map((group, i) => <GroupCard key={i} group={group} />)}
  </Card.Group>
);

GroupCards.propTypes = {
  groups: PropTypes.array
};

const mapStateToProps = ({ groups }) => ({ groups });

export default connect(mapStateToProps)(GroupCards);
