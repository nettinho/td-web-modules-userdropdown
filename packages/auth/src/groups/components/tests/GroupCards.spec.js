import React from "react";
import { shallow } from "enzyme";
import { GroupCards } from "../GroupCards";

describe("<GroupCards />", () => {
  const deleteGroup = jest.fn();
  const groups = [{ id: 1 }, { id: 2 }];
  const props = { groups, deleteGroup };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<GroupCards {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
