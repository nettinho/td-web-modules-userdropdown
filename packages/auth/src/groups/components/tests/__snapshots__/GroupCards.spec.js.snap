// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupCards /> matches the latest snapshot 1`] = `
<CardGroup>
  <Card
    href="/groups/new"
  >
    <CardContent
      textAlign="center"
    >
      <CardHeader>
        <FormattedMessage
          id="groups.actions.create"
          values={Object {}}
        />
      </CardHeader>
      <CardDescription>
        <Icon
          as="i"
          floated="center"
          name="add"
          size="massive"
        />
      </CardDescription>
    </CardContent>
  </Card>
  <Connect(GroupCard)
    group={
      Object {
        "id": 1,
      }
    }
    key="0"
  />
  <Connect(GroupCard)
    group={
      Object {
        "id": 2,
      }
    }
    key="1"
  />
</CardGroup>
`;
